### PR TITLE
libschrift: init at 0.9.1

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

<https://github.com/tomolt/libschrift>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107285` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libschrift</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107285` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libschrift</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107285` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>libschrift</li>
  </ul>
</details>

```
building
cc -Os -std=c99 -pedantic -Wall -Wextra -Wconversion -I./  -c -o schrift.o schrift.c
schrift.c:794:43: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint_fast32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        if ((startCode = getu16(font, startCodes + segIdxX2)) > shortCode)
                         ~~~~~~       ~~~~~~~~~~~^~~~~~~~~~
schrift.c:796:34: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint_fast32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        idDelta = getu16(font, idDeltas + segIdxX2);
                  ~~~~~~       ~~~~~~~~~^~~~~~~~~~
schrift.c:797:52: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint_fast32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        if (!(idRangeOffset = getu16(font, idRangeOffsets + segIdxX2))) {
                              ~~~~~~       ~~~~~~~~~~~~~~~^~~~~~~~~~
schrift.c:803:55: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint_fast32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        idOffset = idRangeOffsets + segIdxX2 + idRangeOffset + 2U * (unsigned int) (shortCode - startCode);
                 ~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
schrift.c:1229:37: warning: implicit conversion loses integer precision: 'unsigned int' to 'uint_fast16_t' (aka 'unsigned short') [-Wconversion]
                        transform_points(outl->numPoints - basePoint, outl->points + basePoint, local);
                        ~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~^~~~~~~~~~~
5 warnings generated.
ar rc libschrift.a schrift.o
ranlib libschrift.a
ld: unknown option: --whole-archive
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107285` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libschrift</li>
  </ul>
</details>

**Comment by sternenseemann** [comment]
@SuperSandro2000 Thank you for trying on darwin! Can you try again, I think I have fixed the linker invocation.

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build libschrift

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107285` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libschrift</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/libschrift/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "c6d20460d6e602e8829d3a227fd7be4c4c3cda86";
+    hash = "sha256-BuTmWaWFZ0DXujlbhbmK3Woit8fR9F4DWmKszHX6gOI=";
+  };
+
+  postPatch = ''
+    substituteInPlace config.mk \
+      --replace "PREFIX = /usr/local" "PREFIX = $out"
+  '';
+
+  buildPhase = "make libschrift.a";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libschrift/default.nix:19]: Please do not overwrite the buildPhase. Use makeFlags instead.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libschrift/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/libraries/libschrift/default.nix]: Resolved.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107285` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libschrift</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107285` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libschrift</li>
</ul>
</details>

**Pull request merged**